src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -851,62 +1233,247 @@ public override IOperation VisitConditionalAccessInstance(IConditionalAccessInst
+                //     body
+                //     goto continue;
+                // }
+                // break:
+
+                AppendNewBlock(@continue);
+
+                if (locals != null)
+                {
+                    EnterRegion(locals);
+                }
+
+                VisitConditionalBranch(operation.Condition, ref @break, sense: operation.ConditionIsUntil);
+
+                VisitStatement(operation.Body);
                 LinkBlocks(CurrentBasicBlock, @continue);
             }
             else
             {
-                LinkBlocks(CurrentBasicBlock, start);
+                // do
+                //   body
+                // while (condition);
+                //
+                // becomes
+                //
+                // start: 
+                // {
+                //   body
+                //   continue:
+                //   GotoIfTrue condition start;
+                // }
+                // break:
+
+                var start = new BasicBlock(BasicBlockKind.Block);
+                AppendNewBlock(start);
+
+                if (locals != null)
+                {
+                    EnterRegion(locals);
+                }
+
+                VisitStatement(operation.Body);
+
+                AppendNewBlock(@continue);
+
+                VisitConditionalBranch(operation.Condition, ref start, sense: !operation.ConditionIsUntil);
+            }
+
+            if (locals != null)
+            {
+                LeaveRegion(locals);
             }
 
-            AppendNewBlock(start);
+            AppendNewBlock(@break);
+
+            return null;
+        }
+
+        public override IOperation VisitTry(ITryOperation operation, int? captureIdForResult)
+        {
+            if (operation.Catches.IsEmpty && operation.Finally == null)
+            {
+                // Malformed node without handlers
+                throw ExceptionUtilities.Unreachable;
+            }
+
+            RegionBuilder tryAndCatchRegion = null;
+            RegionBuilder tryAndCatchTryRegion = null;
+            RegionBuilder tryAndFinallyRegion = null;
+            RegionBuilder tryAndFinallyTryRegion = null;
+
+            if (operation.Finally != null)
+            {
+                tryAndFinallyRegion = new RegionBuilder(ControlFlowGraph.RegionKind.TryAndFinally, _currentRegion);
+                tryAndFinallyTryRegion = new RegionBuilder(ControlFlowGraph.RegionKind.Try, tryAndFinallyRegion);
+                EnterRegion(tryAndFinallyRegion);
+                EnterRegion(tryAndFinallyTryRegion);
+            }
+
+            if (!operation.Catches.IsEmpty)
+            {
+                tryAndCatchRegion = new RegionBuilder(ControlFlowGraph.RegionKind.TryAndCatch, tryAndFinallyTryRegion ?? _currentRegion);
+                EnterRegion(tryAndCatchRegion);
+
+                tryAndCatchTryRegion = new RegionBuilder(ControlFlowGraph.RegionKind.Try, tryAndCatchRegion);
```

Comment:
nit: use the same style as the if above: ie, either create both and then enter both, or create/enter one at a time.